docs/explanation/charm-architecture.md

```diff
@@ -0,0 +1,23 @@
+# Charm architecture
+
+The SAML Integrator charm consists on a charm workload that fetches the SAML metadata details based on the provided configuration and propagates them through the relation.
+
+## Juju events
+
+Accordingly to the [Juju SDK](https://juju.is/docs/sdk/event): "an event is a data structure that encapsulates part of the execution context of a charm".
+
+For this charm, the following events are observed:
+
+1. [upgrade_charm](https://juju.is/docs/sdk/upgrade-charm-event): fired on Kubernetes charms when the unit is undergoing an upgrade. Action: install charm dependencies.
```

Comment:
I think this should be `upgrade-charm` rather than `upgrade_charm` (same for `config-changed` and `saml-relation-joined`). Also, since this charm can be deployed on k8s or machines do we want to remove the "Kubernetes" portion here?

Reply:
Fixed. Thanks